src/aal/aal_arm.h

```diff
@@ -0,0 +1,53 @@
+#pragma once
+
+#if defined(__arch64__)
+#  define SNMALLOC_VA_BITS_64
+#else
+#  define SNMALLOC_VA_BITS_32
+#endif
+
+#include <iostream>
+namespace snmalloc
+{
+  /**
+   * ARM-specific architecture abstraction layer.
+   */
+  class AAL_arm
+  {
+  public:
+    /**
+     * Bitmap of AalFeature flags
+     */
+    static constexpr uint64_t aal_features = IntegerPointers;
+    /**
+     * On pipelined processors, notify the core that we are in a spin loop and
+     * that speculative execution past this point may not be a performance gain.
+     */
+    static inline void pause()
+    {
+      __asm__ volatile("yield");
+    }
+
+    /**
+     * Issue a prefetch hint at the specified address.
+     */
+    static inline void prefetch(void* ptr)
+    {
+      __builtin_prefetch(ptr, 0, 1);
```

Comment:
Is this needed?  If you don't implement `prefetch` then you'll get this implementation from `AAL_Generic`.

Reply:
ok but there is no inheritance unlike pal though.

Reply:
Not exactly.  It's applied centrally.  The per-arch files define an AAL_Arch type, the AAL that the rest of the code uses is defined with the `using Aal = AAL_Generic<AAL_Arch>;` line in aal.h.  We (should) never use `AAL_arm` directly, we will use only `AAL_Generic<AAL_arm>`.  The `prefetch` method in `AAL_Generic` will not call `AAL_arm::prefetch` unless we've disabled builtins for the AAL (in which case, we shouldn't be using it here either).

Reply:
ah right ... I was using wrong compiler gcc 7 ... now the builtins are being detected.